lib/streamlit/elements/widgets/multiselect.py

```diff
@@ -82,15 +82,19 @@ def _check_and_convert_to_indices(
             default_values, "pandas.core.series.Series"
         ):
             default_values = list(cast(Sequence[Any], default_values))
-        elif not default_values or default_values in opt:
-            default_values = [default_values]
-        else:
+        elif (
+            isinstance(default_values, (tuple, set))
```

Comment:
Should we consider generalizing to collections? Any reason not to include dict in this example?

Reply:
I thought a bit today about this, and I agree that we should make `default` support a more generic set of collections. In the best case, it just supports the exactly same formats as `options`. For `options` we use the `ensure_indexable` which we might also want to use in this case. But it would require a bit more refactoring, and I will anyways do refactoring around this logic with the [Support more dataframe formats](https://www.notion.so/snowflake-corp/Support-more-dataframe-formats-Polars-dataframe-interchange-protocol-a125abc972364d0d9c7b47d5b4ef153a) projects. 

So, I think a good way forward is to merge this as a quick fix, and then properly refactor all our iterable data input handling with the project mentioned above. I added this aspect (add more generic support for data types for the multiselect default) to the project page.